File: sources/net.sf.j2s.core/test/dev/j2s/java/security/AccessController.js
Clazz.declarePackage ("java.security");
Clazz.load (["java.security.AccessControlContext"], "java.security.AccessController", null, function () {
c$ = Clazz.declareType (java.security, "AccessController", null, java.security.AccessControlContext);
c$.doPrivileged = Clazz.defineMethod (c$, "doPrivileged", 
function (action) {
return action.run ();
}, "java.security.PrivilegedAction");
c$.getContext = Clazz.defineMethod (c$, "getContext", 
function () {
return  new java.security.AccessController ();
});
Clazz.overrideMethod (c$, "checkPermission", 
function (perm) {
return true;
}, "~O");
c$.doPrivileged = Clazz.defineMethod (c$, "doPrivileged", 
function (action, context) {
return action.run ();
}, "java.security.PrivilegedAction,java.security.AccessControlContext");
});

File: sources/net.sf.j2s.java.core/src/java/security/AccessController.java
package java.security;

public class AccessController implements AccessControlContext {

	// a dummy class
	
	public static <T> T doPrivileged(PrivilegedAction<T> action) {
		return action.run();
	}

	public static void doPrivileged(PrivilegedExceptionAction<Void> privilegedExceptionAction) {
		try {
			privilegedExceptionAction.run();
		} catch (Exception e) {
		}		
	}

	public static AccessControlContext getContext() {
		return new AccessController();
	}

	@Override
	public boolean checkPermission(Object perm) {
		// no access checking in JavaScript
		return true;
	}

  public static <T> T doPrivileged(PrivilegedAction<T> action,
      AccessControlContext context) {
  	return action.run();
  }

}
